test/IntegrationTests/MSTest.Acceptance.IntegrationTests/InitializeAndCleanupTimeoutTests.cs

```diff
@@ -286,6 +370,62 @@ public static async Task ClassInit(TestContext testContext)
         }
     }
 
+    $TimeoutAttribute$
+    [TestInitialize]
+    public async Task TestInit()
+    {
+        if (Environment.GetEnvironmentVariable("LONG_WAIT_TESTINIT") == "1" || Environment.GetEnvironmentVariable("TIMEOUT_TESTINIT") == "1")
+        {
+            await Task.Delay(10_000);
+        }
+        else
+        {
+            await Task.CompletedTask;
+        }
+    }
+
+    $TimeoutAttribute$
+    [TestCleanup]
+    public async Task TestCleanupMethod()
+    {
+        if (Environment.GetEnvironmentVariable("LONG_WAIT_TESTCLEANUP") == "1" || Environment.GetEnvironmentVariable("TIMEOUT_TESTCLEANUP") == "1")
+        {
+            await Task.Delay(10_000);
+        }
+        else
+        {
+            await Task.CompletedTask;
+        }
+    }
+
+    $TimeoutAttribute$
+    [ClassCleanup]
+    public static async Task ClassCleanupMethod()
```

Comment:
can you move the cleanup method after the relative init for clarity?

I got the reason for the inner structure but it's harder to review imo